File: bookcatalogue/src/main/java/com/example/bookcatalogue/model/Author.java
package com.example.bookcatalogue.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Clob;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Table(name = "AUTHOR")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    private int version;

    @Column(name = "author_name", nullable = false)
    private String name;

    @Column(nullable = false)
    private String nationality;

    @Column(nullable = false, columnDefinition = "text")
    private String bio;

    @JsonIgnore
    @ManyToMany(mappedBy = "authors")
    private Set<Book> books = new HashSet<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Author author = (Author) o;
        return Objects.equals(name, author.name) &&
            Objects.equals(nationality, author.nationality) &&
            Objects.equals(bio, author.bio);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), name, nationality, bio);
    }
}

File: ng-bookcatalogue/src/app/classes/author.ts
export class Author {
  private _id: number;
  private _name: string;
  private _nationality: string;
  private _bio: string;

  public constructor(id: number, name: string, nationality: string, bio: string) {
    this._id = id;
    this._name = name;
    this._nationality = nationality;
    this._bio = bio;
  }


  get id(): number {
    return this._id;
  }

  public get name(): string {
    return this._name;
  }

  public get nationality(): string {
    return this._nationality;
  }

  public get bio(): string {
    return this._bio;
  }
}
